### PR TITLE
[ENH] Datalad documentation, archiving The Way, updating information on Jekyll versions for GH page building

### PR DESCRIPTION
**Pull request by kahinimehta**

### CODE REVIEW

File: docs/Contributing/documentation_guidelines.md [hunk 1]
```diff
@@ -122,26 +122,32 @@ Next, make sure you have `Ruby` installed; the best way to check this is to do `
 To install ruby, just do:
 
 ```
-brew install ruby
+brew install ruby@3.2
```
Comment (tsalo) [inline docs/Contributing/documentation_guidelines.md]: Did this work for you?

Comment (tsalo) [reply docs/Contributing/documentation_guidelines.md]: BTW I rename this file to `docs/documentation/documentation_guidelines.md` in #79, so we'll have conflicts to deal with.

File: docs/DataWorkflows/DataCycle.md [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+title: Data Cycle
+parent: Data Workflows
+nav_order: 1
+has_toc: true
+---
+# The Data Cycle
+
+The data cycle includes the following steps: 
+1. Fetching raw data
+3. Curation of datasets
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md]:
```suggestion
3. Dataset curation
```

File: docs/DataWorkflows/DataCycle.md [hunk 3]
```diff
@@ -0,0 +1,671 @@
...
+2. Fetching the data onto the right cluster
+3. Unzipping the data (as needed)
+4. Processing the data!
+5. Sharing the data
+
+Each of these steps is detailed here: 
+
+# Fetching Your Data
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md:20]: I propose including datalad here. That's how I grab OpenNeuro datasets.

File: docs/DataWorkflows/DataCycle.md [hunk 4]
```diff
@@ -0,0 +1,671 @@
...
+
+# with fw-heudiconv
+fw-heudiconv-export --project <PROJECT_NAME> --subject <SUBJECTS_FILTER> --session <SESSION_FILTER> --folders <LIST_OF_BIDS_FOLDERS>
+```
+
+Try `fw-heudiconv-export -h` for more info.
+
+## Globus
+
+##### Note: this section has not been tested in a long time
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md]:
```suggestion
{: .warning-title }
> Warning
>
> This section has not been tested in a long time.
```

Red admonition box to make it pop.

File: docs/DataWorkflows/DataCycle.md [hunk 5]
```diff
@@ -0,0 +1,671 @@
...
+
+4. Visit [https://docs.globus.org/how-to/get-started/](https://docs.globus.org/how-to/get-started/) to access the File Manager, as in the Local Disk instructions, to start transferring data.
+
+
+# Curating BIDS Datasets
+
+
+BIDS curation can be a frustrating process. This guide describes best practices for
+the curation process using a local filesystem. We divide this process into multiple
+stages.
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md]:
```suggestion
BIDS curation can be a frustrating process. 
This guide describes best practices for the curation process using a local filesystem. 
We divide this process into multiple stages.
```

I find version control on markdown files easier if I split lines based on sentences and commas. No need to adopt this if you don't like it.

File: docs/DataWorkflows/DataCycle.md [hunk 6]
```diff
@@ -0,0 +1,671 @@
...
+changes we might make to it. The person in charge of organizing a data set
+should document any changes they make in a *Data Narrative* along with any
+relevant scripts used to a git-tracked repository.
+
+Create a project root directory. Initially we suggest keeping one copy of your
+initial data that will remain untouched as a backup during early stages of
+the workflow. This is not necessary if your data is archived elsewhere online.
+In the project root create a directory structure similar to this:
+
+```
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md:159]: I think it would be good to link out to [`paper-template`](https://github.com/PennLINC/paper-template) for this. If that template doesn't have a good proposed organization, we can change it.

File: docs/DataWorkflows/DataCycle.md [hunk 7]
```diff
@@ -0,0 +1,671 @@
...
+    └── BIDS
+```
+
+You can add git like so:
+
+```bash
+$ git init
+```
+
+And remember, if you explicitly want to ignore something from tracking (with either `git` or `datalad`), add that path to a `.gitnignore` file ([see here](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) and [here](https://handbook.datalad.org/en/latest/beyond_basics/101-179-gitignore.html)).
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md]:
```suggestion
And remember, if you explicitly want to ignore something from tracking (with either `git` or `datalad`), add that path to a `.gitignore` file ([see here](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) and [here](https://handbook.datalad.org/en/latest/beyond_basics/101-179-gitignore.html)).
```

File: docs/DataWorkflows/DataCycle.md [hunk 8]
```diff
@@ -0,0 +1,671 @@
...
+
+### Add NIfTI information to the sidecars
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md:210]: Probably best to link out to CuBIDS's documentation for this, since CuBIDS _should_ have this info in one of its tutorials (if it doesn't, we should add it).

File: docs/DataWorkflows/DataCycle.md [hunk 9]
```diff
@@ -0,0 +1,671 @@
...
+[BABS](https://pennlinc-babs.readthedocs.io/en/latest/) is a quick and easy tool for processing pipelines via [Datalad](./Datalad.md). The documentation is thorough, with information on setting up, installing, and running BABS on data. An example walkthrough is also available via the documentation. 
+
+# Sharing your data
+{: .no_toc}
+
+Collaboration is an important part of our work. Sharing the input data and outputs/derivatives of a project is a critical step in the scientific process, and so doing it both accurately and efficiently is a high priority. Overall, remember that this step can be a strain, especially when dealing with external collaborators or systems that are foreign to you. Be patient, and remember that we have to balance accuracy/reproducibility with efficiency/speed.
+
+Below are a few example scenarios and what best recommendations we have for sharing data, particularly outputs from BABS
+
+## Datalad to Datalad
```
Comment (tsalo) [inline docs/DataWorkflows/DataCycle.md:633]: Can you add a section on Datalad to OpenNeuro? See #60. Can be left empty here, but will remind us to fill it out at some point.

File: docs/DataWorkflows/FetchingYourData.md [hunk 10]
```diff
@@ -1,8 +1,8 @@
 ---
 layout: default
 title: Getting Static Data from PMACS
```
Comment (tsalo) [inline docs/DataWorkflows/FetchingYourData.md]: I would rename the whole file to reflect that this is specific to PMACS.

File: docs/archive/index.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (tsalo) [inline docs/archive/index.md]: It's odd that this file would be created. Pretty sure someone else created it earlier today. Is it possible that there are two similarly-named files/folders?

**Review by tsalo — COMMENTED** [review]
It's looking really good! I have a few thoughts though. Also, I think the changes to jekyll files might be specific to the older version of just-the-docs (i.e., pre-#77). Just want to make sure those are good additions.

File: docs/archive/index.md (hunk 11 above)
Comment (tsalo) [reply docs/archive/index.md]: Ah yes @gshafiei used `Archive` instead. I would rename your folder so there's only one.

File: docs/Contributing/documentation_guidelines.md (hunk 1 above)
Comment (kahinimehta) [reply docs/Contributing/documentation_guidelines.md]: okay, makes sense - yes that worked for me!

File: docs/DataWorkflows/FetchingYourData.md [hunk 12]
```diff
@@ -81,8 +81,6 @@ sessions:
 $ echo 'module load git' >> ~/.bashrc
 ```
 
-
-
 # Clone the static data
 
 In order to get the data to your computer, you need to ensure 3 things:
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: I would rephrase saying: "in order to get the data to your computer or in your CUBIC project directory"

Comment (joellebagautdinova) [reply docs/DataWorkflows/FetchingYourData.md]: For the list below, I'd also specify: 
2. You have datalad, git and git-annex installed on your computer or on your CUBIC project directory (see the section on installing [Datalad](/docs/DataWorkflows/Datalad.md)  
an then add an extra point:
5. If you're planning to download data into your CUBIC project directory, you are logged in to CUBIC project user (see instructions [here](/docs/cubic/cubic.md)

File: docs/DataWorkflows/FetchingYourData.md [hunk 13]
```diff
@@ -95,7 +93,7 @@ In order to get the data to your computer, you need to ensure 3 things:
 
 If these conditions are met, you're ready to access some data! Suppose I'd like
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: For more clarity, I'd break this with a subtitle up in the following way: 
If these conditions are met, you're ready to access some data!
Generally, datasets can be in two different formats: 
- **unzipped**: this type of dataset allows to access all different subdirectories and files of all participants directly
- **zipped**: this type of dataset has one zip file per participant. This allows more efficient storing, and requires a few extra steps to figure out what kinds of files are available and to download them. 

Ways of looking at and then getting each type of dataset are explained below.

## Getting an unzipped dataset
Suppose I'd like to see the regional time series data for CCNP.
[etc.]

File: docs/DataWorkflows/FetchingYourData.md [hunk 14]
```diff
@@ -143,8 +141,7 @@ sub-colornest002
 sub-colornest195
 ```
 
-This directory contains all the outputs from xcp_d for each subject. Now, let's see if we can find the
-actual file we want to get for each subject. We can see all the files for a single subject with `find`
+This directory contains all the outputs from xcp_d for each subject. Now, let's see if we can find the actual file we want to get for each subject. We can see all the files for a single subject with `find`
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: I would add a subtitle in the text above, in this way:

This directory contains all the outputs from xcp_d for each subject. 

### Figuring out what data is available (single participant)
Now, let's see if we can find the actual file we want to get for each subject. We can see all the files for a single subject with `find`:

File: docs/DataWorkflows/FetchingYourData.md [hunk 15]
```diff
@@ -143,8 +141,7 @@ sub-colornest002
...
 
 ```
 $ find sub-colornest001
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: I'd change this sentence: 

Old: You may be asked for your password if you haven't [set up an ssh key](#set

New: "You will be asked for your password."

Comment (joellebagautdinova) [reply docs/DataWorkflows/FetchingYourData.md]: Then I'd add a new subsection: 
I see that the atlas I want (Gordon) can be found for each subject with a pattern. 

### Getting files for the entire dataset
To actually transfer these files for all participants, I need to tell datalad to fetch them from PMACs. This is done using wildcards (*) with the following command:

File: docs/DataWorkflows/FetchingYourData.md [hunk 16]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
 analysis will be fetched if you need to
 [access the data again](#clone-the-static-data).
 
+# Unzipping your data
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: I'd change this header to: 

## Getting an zipped dataset

File: docs/DataWorkflows/FetchingYourData.md [hunk 17]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+
+Data on PMACS might be zipped, or unzipped. In the case of unzipped data, you should be good to go following the guidelines on finding files you need [here](./FetchingYourData.md)! If your data is zipped, you might need a little more work.
+
+The following walkthrough should help you unzip your data. These lines of code can be customized per your needs to unzip some or all subjects and then some or all files:
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: I'd replace the sentence as follows: 
Old: "Data on PMACS might be zipped, or unzipped. In the case of unzipped data, you should be good to go following the guidelines on finding files you need [here](./FetchingYourData.md)! If your data is zipped, you might need a little more work.
The following walkthrough should help you unzip your data. These lines of code can be customized per your needs to unzip some or all subjects and then some or all files:"

New: The following walkthrough will cover how to work with a zipped dataset.

File: docs/DataWorkflows/FetchingYourData.md [hunk 18]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
 analysis will be fetched if you need to
 [access the data again](#clone-the-static-data).
 
+# Unzipping your data
+
+Data on PMACS might be zipped, or unzipped. In the case of unzipped data, you should be good to go following the guidelines on finding files you need [here](./FetchingYourData.md)! If your data is zipped, you might need a little more work.
+
+The following walkthrough should help you unzip your data. These lines of code can be customized per your needs to unzip some or all subjects and then some or all files: 
+
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: similar to the above organization I proposed, here I'd add a subsection: 

### Figuring out what data is available (single participant)

File: docs/DataWorkflows/FetchingYourData.md [hunk 19]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+# get the file for one specific participant
+datalad get $file -J 8 # runs the job on 8 threads
+```
+
+The output of datalat get will look like this: 
+
+```bash
+It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
+Total:   0%|                                                                                | 0.00/1.19G [00:00<?, ? Bytes/s]
+dumbledore@bblsub.pmacs.upenn.edu's password:
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: if we want to follow the structure for the unzipped dataset, here bblsub should be replaced with [login node name]
(and wherever else is relevant below)

File: docs/DataWorkflows/FetchingYourData.md [hunk 20]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+datalad drop --nocheck $file
+```
+
+This is what the output should look like: 
+
+```bash
+It is highly recommended to configure Git before using DataLad. Set both 'user.name' and 'user.email' configuration variables.
+drop(ok): sub-NevilleLongbottom_fmriprep-20.2.3.zip (file)
+```
+
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: Add subsection header here:

### Getting files for the entire dataset

File: docs/DataWorkflows/FetchingYourData.md [hunk 21]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+```bash
+#!/bin/bash
+# conda activate dlad
+
+# clone static data of interest from PMACS, here LINC_PNC#~FMRIPREP_zipped, 
+# but can be replaced with anything in the "Clone URL" column here https://pennlinc.github.io/docs/DataTasks/AvailableStaticData/
+datalad clone ria+ssh://dumbledore@bblsub.pmacs.upenn.edu:/static/LINC_PNC#~FMRIPREP_zipped /cbica/projects/hogwarts/data/PNC/PNC_fMRIPrep_v20_2_3/datalad
+
+# do this so that you don't have to enter your pmacs password 1 million times when datalad getting files
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: Not sure if this should be on there given that we don't recommend using the ssh keys?

File: docs/DataWorkflows/FetchingYourData.md [hunk 22]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+	# # drop the file
+	datalad drop --nocheck $file
+
+done
+
+# -------------------------------------------------
+#### get zip files for the entire PNC dataset #####
+# -------------------------------------------------
+
+# Note: if this requires inputting the password every time, may need to switch to python and use glob patterns
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: can remove this line, not needed here

Note: if this requires inputting the password every time, may need to switch to python and use glob patterns

File: docs/DataWorkflows/FetchingYourData.md [hunk 23]
```diff
@@ -246,6 +243,212 @@ Critically, the reference to the EXACT version of the file you used in your
...
+for file in *zip ; do
+
+	sub=${file%_*}
+	datalad get $file -J 8
+
+ 	# unzip the zip 
+	unzip "/cbica/projects/hogwarts/data/PNC/PNC_fMRIPrep_v20_2_3/zipped/$file" -d /cbica/projects/hogwarts/data/PNC/PNC_fMRIPrep_v20_2_3/unzipped/
+
+ 	# unzip specific files only (needs tweatking if want specific files, example here)
+ 	# unzip -j "$file" "xcp_abcd/$sub/ses-PNC1/func/*ses-PNC1_task-rest_acq-singleband_space-fsLR_den-91k_desc-alff*" -d /cbica/projects/hogwarts/data/PNC/PNC_fMRIPrep_v20_2_3/$sub
```
Comment (joellebagautdinova) [inline docs/DataWorkflows/FetchingYourData.md]: This should be updated to: 

unzip -j "$file" "fmriprep/$sub/ses-PNC1/func/${sub}_ses-PNC1_task-rest_acq-singleband_space-fsLR_den-91k_bold.dtseries*" -d /cbica/projects/hogwarts/data/PNC/PNC_fMRIPrep_v20_2_3/$sub
(still comment it out though to highlight that this is an alternative to the above unzip for all participant files

also there is a typo in tweaking lol

**Review by joellebagautdinova — CHANGES_REQUESTED** [review]

File: docs/DataWorkflows/DataCycle.md (hunk 3 above)
Comment (kahinimehta) [reply docs/DataWorkflows/DataCycle.md:20]: added

File: docs/DataWorkflows/DataCycle.md (hunk 8 above)
Comment (kahinimehta) [reply docs/DataWorkflows/DataCycle.md:210]: Yep, we will have to add it

File: docs/DataWorkflows/DataCycle.md (hunk 9 above)
Comment (kahinimehta) [reply docs/DataWorkflows/DataCycle.md:633]: added the section + linked to docs on how to do it

File: docs/DataWorkflows/FetchingYourData.md (hunk 10 above)
Comment (kahinimehta) [reply docs/DataWorkflows/FetchingYourData.md]: done

File: docs/archive/index.md (hunk 11 above)
Comment (kahinimehta) [reply docs/archive/index.md]: Done!
